packages/ckeditor5-package-generator/lib/utils/choose-package-manager.js

```diff
@@ -5,24 +5,27 @@
 
 import inquirer from 'inquirer';
 import isYarnInstalled from './is-yarn-installed.js';
+import isPnpmInstalled from './is-pnpm-installed.js';
 
 /**
  * @param {Boolean} useNpm
  * @param {Boolean} useYarn
- * @returns {Promise<'npm'|'yarn'>}
+ * @param {Boolean} usePnpm
+ * @returns {Promise<'npm'|'yarn'|'pnpm'>}
  */
-export default async function choosePackageManager( useNpm, useYarn ) {
+export default async function choosePackageManager( useNpm, useYarn, usePnpm ) {
 	const yarnInstalled = isYarnInstalled();
+	const pnpmInstalled = isPnpmInstalled();
```

Comment:
I'm thinking whether we should remove checking if `yarn` or `pnpm` is installed. If they are not, an error would be thrown anyway, and it should be pretty obvious that they should be installed.

We could remove two functions: `isYarnInstalled`, `isPnpmInstalled`, and simplify the logic.

Reply:
We did it to detect the available package managers. Unfortunately, I am unsure whether it was necessary or if we used a _fancy_ switcher.

Reply:
Removed this code in [1188077](https://github.com/ckeditor/ckeditor5-package-generator/pull/249/commits/11880771ea50faae55c30c026890145544fae1f9).

Reply:
I reverted the removal of detection if package manager is installed in [279e013](https://github.com/ckeditor/ckeditor5-package-generator/pull/249/commits/279e01341bc587837a119c4298d8bf0a1ff0d9b0).

Reply:
We need to think now, whether we should display package managers in the prompt if they are not installed @pomek .